Description: Make lightspark build under LLVM3.5
 .
 Also add -lpthread and -ltinfo somehow missing.
Author: Eugene Shatokhin
Origin: https://abf.rosalinux.ru/import/lightspark/commit/8cc84aa57a33b33ed9ef83c9bc9065d563a40a14
Bug-Debian: http://bugs.debian.org/763235
Last-Update: 2014-10-16
--- a/src/scripting/abc.cpp
+++ b/src/scripting/abc.cpp
@@ -35,7 +35,7 @@
 #include <llvm/Target/TargetSelect.h>
 #endif
 #include <llvm/Target/TargetOptions.h>
-#include <llvm/Analysis/Verifier.h>
+#include <llvm/IR/Verifier.h>
 #include <llvm/Transforms/Scalar.h> 
 #include "logger.h"
 #include "swftypes.h"
@@ -1487,7 +1487,7 @@
 		assert_and_throw(th->ex);
 
 		th->FPM=new llvm::FunctionPassManager(th->module);
-		th->FPM->add(new llvm::DataLayout(*th->ex->getDataLayout()));
+		th->module->setDataLayout(new llvm::DataLayout(*th->ex->getDataLayout()));
 #ifdef EXPENSIVE_DEBUG
 		//This is pretty heavy, do not enable in release
 		th->FPM->add(llvm::createVerifierPass());
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -140,7 +140,7 @@
 ENDIF (CMAKE_COMPILER_IS_GNUCC)
 
 TARGET_LINK_LIBRARIES(spark ${CAIRO_LIBRARIES} ${ZLIB_LIBRARIES}
-	${Boost_LIBRARIES} ${LLVM_LIBS_CORE} ${LLVM_LIBS_JIT} ${LLVM_LDFLAGS}
+	${Boost_LIBRARIES} ${LLVM_LIBS_CORE} ${LLVM_LIBS_JIT} ${LLVM_LDFLAGS} -lpthread -ltinfo
 	${OPTIONAL_LIBRARIES} ${GTK_LIBRARIES} ${FREETYPE_LIBRARIES} ${JPEG_LIBRARIES} ${PNG_LIBRARIES}
 	${PCRE_LIBRARIES} ${GLIBMM_LIBRARIES} ${GMODULE_LIBRARIES}
 	${XMLPP_LIBRARIES} ${CMAKE_DL_LIBS} ${EXTRA_LIBS_LIBRARIES}
